ufo2fontir/src/source.rs

```diff
@@ -845,6 +866,107 @@ impl Work<Context, WorkError> for FeatureWork {
     }
 }
 
+impl Work<Context, WorkError> for KerningWork {
+    /// See <https://github.com/googlefonts/ufo2ft/blob/3e0563814cf541f7d8ca2bb7f6e446328e0e5e76/Lib/ufo2ft/featureWriters/kernFeatureWriter.py#L302-L357>
+    fn exec(&self, context: &Context) -> Result<(), WorkError> {
+        debug!("Kerning for {:#?}", self.designspace_file);
+
+        let designspace_dir = self.designspace_file.parent().unwrap();
+        let static_metadata = context.get_final_static_metadata();
+        let master_locations = master_locations(&static_metadata.axes, &self.designspace.sources);
+
+        let mut kerning = Kerning::default();
+
+        // Pass 1: find all the groups used by any source
+        // Woe unto ye who define the same group but inconsistently; we effectively union the members
+        for source in self.designspace.sources.iter() {
```

Comment:
similarly to how we do for the global metrics, we need to skip for kerning those sources that have not-None `layer` attribute and that only contribute glyph data:

https://github.com/googlefonts/fontc/blob/04465968e5a7d68dfb1c80b058ccfc2d421286b7/ufo2fontir/src/source.rs#L752-L759

ufo2ft similarly doesn't compile features for sources whose layerName is not None

Reply:
Ew. Fixed.